rfcs/020-automated-release-process.md

```diff
@@ -0,0 +1,84 @@
+- Feature Name: Automated Release Process
+- Start Date: 2020-12-19
+- RFC PR: [amundsen-io/rfcs#20](https://github.com/amundsen-io/rfcs/pull/20)
+- Amundsen Issue: [amundsen-io/amundsen#0000](https://github.com/amundsen-io/amundsen/issues/0000) (leave this empty for now)
+
+# Automated Release Process
+
+## Summary
+
+This RFC proposes to automate release with github actions.
+
+## Motivation
+
+Currently release process requires manual release creation and changelog rendering (by going through list of commits).
+
+The motivation behind this RFC is to automate, unify and simplify release process.
+
+## Guide-level Explanation (aka Product Details)
+
+Having enforced naming convention for MRs we can leverage this to automatically categorize MRs and render them in
+changelog. By using automated github actions we can remove manual step from release process (releasing, rendering changelog)
+and replace it with three operations:
+
+- creating a git tag triggers a release
+- periodical releases every month
+- pre-releases with each change added to master
+
+## Reference-level Explanation (aka Technical Details)
+
+The plan for each release option is:
+
+### Continuous Pre-releases
+
+- Implement a new github action called `Pre-releases` that will trigger on every PR merged into master
```

Comment:
maybe to reduce no of pre-releases we could do them like monthly but with weekly granularity ?